blivet/devicefactory.py

```diff
@@ -1471,6 +1471,10 @@ def _get_child_factory_kwargs(self):
 
     def _get_new_device(self, *args, **kwargs):
         """ Create and return the factory device as a StorageDevice. """
+
+        name = kwargs.pop("name", "")
+        kwargs["name"] = self.storage.unique_device_name(name, parent=self.vg, name_set=True)
+
```

Comment:
No change to `LVMFactory._set_name`?

Reply:
The original name in `LVMFactory._set_name` is unique so it simply refuses to rename the device to the new non-unique name, which makes sense. This is a special case where the "original" name is not unique and there is no new name so I decided to make it unique only in this one special case as a exception.